### PR TITLE
Updated error message when hit "not found error" in gitprovider

### PR DESCRIPTION
**Pull request by josecordaz**
Closes: #1048 

**What changed?**
Updated error message.
From:
```
Error: failed to get git clients: error setting up deploy keys: error uploading deploy key: error uploading deploy key multiple errors occurred:
- POST https://api.github.com/repos/stefanprodan/podinfo/keys: 404 Not Found []
- the requested resource was not found
```
To: 
```
Error: failed to get git clients: error setting up deploy keys: error uploading deploy key: no permissions to access this repository or repository doesn't exists
```

**Why?**
There was confusion when distinguishing between if there were not enough permissions and if the repository does not exist.

**How did you test it?**
Added unit tests

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: pkg/gitproviders/provider_org_test.go [hunk 1]
```diff
@@ -137,6 +137,13 @@ var _ = Describe("Org Provider", func() {
 			Expect(err.Error()).Should(ContainSubstring("error uploading deploy key"))
 		})
 
+		It("return error when it fails to upload a deploy key", func() {
```
Comment (jpellizzari) [inline pkg/gitproviders/provider_org_test.go:140]: :+1:

File: pkg/gitproviders/provider_user_test.go [hunk 2]
```diff
@@ -94,7 +94,7 @@ var _ = Describe("User Provider", func() {
 			deployKeyClient.GetReturns(nil, gitprovider.ErrNotFound)
 
 			res, err := userProvider.DeployKeyExists(ctx, repoUrl)
-			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ShouldNot(HaveOccurred())
```
Comment (jpellizzari) [inline pkg/gitproviders/provider_user_test.go:97]: Why the change to this?

**Review by jpellizzari — APPROVED** [review]

File: pkg/gitproviders/provider_user_test.go (hunk 2 above)
Comment (josecordaz) [reply pkg/gitproviders/provider_user_test.go:97]: I'll revert this, I modified it twice and when I tried to put it back to the previous state forgot it had `ToNot` instead of `ShouldNot`.

File: pkg/gitproviders/provider.go [hunk 3]
```diff
@@ -89,7 +91,11 @@ func deployKeyExists(ctx context.Context, repo gitprovider.UserRepository) (bool
 func uploadDeployKey(ctx context.Context, repo gitprovider.UserRepository, deployKeyInfo gitprovider.DeployKeyInfo) error {
 	_, err := repo.DeployKeys().Create(ctx, deployKeyInfo)
 	if err != nil {
-		return fmt.Errorf("error uploading deploy key %s", err)
+		if errors.Is(err, gitprovider.ErrNotFound) {
```
Comment (luizbafilho) [inline pkg/gitproviders/provider.go:94]: in general in Go we should use else clauses unless strictly necessary, which is not the case here since we call `return` in the `if`, so a suggestion would be changing to:
```go
if errors.Is(err, gitprovider.ErrNotFound) {
	return RepositoryNoPermissionsOrDoesNotExistError
}

return fmt.Errorf("error uploading deploy key %s", err)
```

File: pkg/gitproviders/provider.go [hunk 4]
```diff
@@ -25,6 +25,8 @@ const (
 	defaultTimeout = time.Second * 30
 )
 
+var RepositoryNoPermissionsOrDoesNotExistError = errors.New("no permissions to access this repository or repository doesn't exists")
```
Comment (luizbafilho) [inline pkg/gitproviders/provider.go:28]: major: errors are prefixed with the `Err` like in `gitprovider.ErrNotFound` and the error types are suffixed with the `Error`

**Review by luizbafilho — CHANGES_REQUESTED** [review]

**Review by luizbafilho — APPROVED** [review]
